### DISCUSSION TITLE
AI Thinker Audiokit V 2.2 with a 2974 A1S

### DISCUSSION THREAD

**Discussion by gerthalberg** [pschatzmann/arduino-audio-tools#299] [Q&A] [answered]
I have got the audiokit to audiokit I2S working with guitar as input and speaker output. I have also tried the ICYstream and it's working as well. I'm using the 2954 (board option 7) as it has the right pins for this version. 5, 25,26,35 and 18, 23. I have tried another library and this confirms that the SD card pins are 14, 15 and 2 for MMC. In order to use this very nice library I have however to set #define AUDIOKIT_SETUP_SD 0 and I have to delete the cfg.sd_active = false; line from the setup() routine. It's not a major thing for me that there are some quirks with the sd but if you want to adopt the lib to this particular version I'm available to give some assistance given the right guidance.

Even though I set the AUDIOKIT_SETUP_SD 0 I still get this in the log AudioKit.h : 46 - sd_active = true.

**Comment by pschatzmann** [comment, marked as answer]
The SD pins are the same across all boards.
Here is a SDMMC example that is working for me w/o changing any code
https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-audiokit/player-sdmmc-audiokit

ps. you can't define AUDIOKIT_SETUP_SD in your sketch, but you would need to change this in the AudioConfig.h. Therefore I prefer to use the logic that you find in the example:
The cfg.sd_active = false is deactivating the spi sd setup. This is however not absolutely necessary because it would work even w/o this.

 For SDMMC make sure that the pins are set to on, on, on, on, on
